code/ui/components/src/components/Button/Button.tsx

```diff
@@ -1,29 +1,121 @@
-import type { FC, ComponentProps, ReactNode } from 'react';
-import React, { forwardRef } from 'react';
+import type { ButtonHTMLAttributes, SyntheticEvent } from 'react';
+import React, { forwardRef, useEffect, useState } from 'react';
 import { styled } from '@storybook/theming';
 import { darken, lighten, rgba, transparentize } from 'polished';
+import { Slot } from '@radix-ui/react-slot';
+
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  asChild?: boolean;
+  size?: 'small' | 'medium';
+  padding?: 'small' | 'medium';
+  variant?: 'outline' | 'solid' | 'ghost';
+  onClick?: (event: SyntheticEvent) => void;
+  disabled?: boolean;
+  active?: boolean;
+  onClickAnimation?: 'none' | 'rotate360' | 'glow' | 'jiggle';
 
-const ButtonWrapper = styled.button<{
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
```

Comment:
"This API is not used anymore." is not really true, since it is still respected - I think you can remove this sentence.

We also want to point the user towards the alternative that they can use instead.

Here's a format I would propose for all of these:

```ts
/**
 * @deprecated Use {@link asChild} instead
 */
```

It would also be nice to split `ButtonProps` into two types, `ButtonProps` and `DeprecatedButtonProps`, because it would make it more apparent to users what is being changed, but it's just my personal preference.

Reply:
I improved the depreciated naming based on your feedbacks. I'm using `ButtonProps` in multiple places so I might keep it like this for now.